searchlib/src/vespa/searchlib/index/field_length_calculator.h

```diff
@@ -0,0 +1,41 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include <atomic>
+#include <algorithm>
+
+namespace search::index {
+
+/**
+ * Class used to calculate average field length, with a bias towards
+ * the latest field lengths when MAX_NUM_SAMPLES samples have been reached.
+ */
+class FieldLengthCalculator {
+    std::atomic<double>   _average_field_length;
+    uint32_t              _num_samples;     // Capped by _max_num_samples
+    static constexpr uint32_t    MAX_NUM_SAMPLES = 100000;
```

Comment:
I think we should provide this in the constructor, with default of 100k if not specified. Then we can choose a lower number in average_until_max_num_samples test.